pages/Receive.qml

```diff
@@ -75,14 +75,182 @@ Rectangle {
+                font.pixelSize: 17
                 textFormat: Text.RichText
-                text: qsTr("Addresses") + translationManager.emptyString
+                color: MoneroComponents.Style.defaultFontColor
+                themeTransition: false
+            }
+
+            MoneroComponents.TextPlain {
+                id: selectedAddressDrescription
+                Layout.alignment: Qt.AlignHCenter
+                Layout.maximumWidth: 220
+                text: "(" + qsTr("no label") + ")" + translationManager.emptyString
+                wrapMode: Text.WordWrap
+                font.family: MoneroComponents.Style.fontRegular.name
+                font.pixelSize: 17
+                textFormat: Text.RichText
+                color: selectedAddressDrescriptionMouseArea.containsMouse ? MoneroComponents.Style.orange : MoneroComponents.Style.dimmedFontColor
+                themeTransition: false
+                MouseArea {
+                    id: selectedAddressDrescriptionMouseArea
+                    visible: subaddressListView.currentIndex > 0
+                    hoverEnabled: true
+                    anchors.fill: parent
+                    cursorShape: Qt.PointingHandCursor
+                    onClicked: {
+                        renameSubaddressLabel(appWindow.current_subaddress_table_index);
+                    }
+                }
+            }
+
+            MoneroComponents.TextPlain {
+                id: selectedAddress
+                Layout.alignment: Qt.AlignHCenter
+                Layout.maximumWidth: 300
+                text: appWindow.current_address ? appWindow.current_address : ""
+                horizontalAlignment: TextInput.AlignHCenter
+                wrapMode: Text.Wrap
+                textFormat: Text.RichText
+                color: selectedAddressMouseArea.containsMouse ? MoneroComponents.Style.orange : MoneroComponents.Style.defaultFontColor
+                font.pixelSize: 15
+                font.family: MoneroComponents.Style.fontRegular.name
+                themeTransition: false
+                MouseArea {
+                    id: selectedAddressMouseArea
+                    hoverEnabled: true
+                    anchors.fill: parent
+                    cursorShape: Qt.PointingHandCursor
+                    onClicked: {
+                        clipboard.setText(appWindow.current_address);
+                        var selectedAddressLabel = appWindow.currentWallet.getSubaddressLabel(appWindow.currentWallet.currentSubaddressAccount, subaddressListView.currentIndex);
+                        if (selectedAddressLabel == "") {
+                            appWindow.showStatusMessage(qsTr("Address copied to clipboard") + translationManager.emptyString, 3);
+                        } else {
+                            appWindow.showStatusMessage(qsTr("Address (%1) copied to clipboard").arg(selectedAddressLabel) + translationManager.emptyString, 3);
+                        }
```

Comment:
```suggestion
                        appWindow.showStatusMessage(qsTr("Address copied to clipboard") + translationManager.emptyString, 3);
```

Just my personal opinion. The label is directly above the address you are clicking anyway and the text is only visible 3 seconds so we want to keep the information as concise as possible.